server/src/test/resources/smoke-test.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+# Variable to track whether shutdown has been called
+SHUTDOWN_CALLED=0
+
+# Function to shut down the background processes
+shutdown() {
+    if [[ $SHUTDOWN_CALLED -eq 1 ]]; then
+        return
+    fi
+    SHUTDOWN_CALLED=1
+
+    echo "Shutting down background processes..."
+
+    if [[ -n "$PRODUCER_PID" ]]; then
+        if kill -0 $PRODUCER_PID 2>/dev/null; then
+            kill $PRODUCER_PID
+            wait $PRODUCER_PID 2>/dev/null
+            echo "Producer process $PRODUCER_PID terminated."
+        else
+            echo "Error: Producer process $PRODUCER_PID has already terminated."
+            echo "Producer logs:"
+            cat producer.log
+            exit 1
+        fi
+    fi
+
+    if [[ -n "$CONSUMER_PID" ]]; then
+        if kill -0 $CONSUMER_PID 2>/dev/null; then
+            kill $CONSUMER_PID
+            wait $CONSUMER_PID 2>/dev/null
+            echo "Consumer process $CONSUMER_PID terminated."
+        else
+            echo "Error: Consumer process $CONSUMER_PID has already terminated."
+            echo "Consumer logs:"
+            cat consumer.log
+            exit 1
+        fi
+    fi
+}
+
+# Trap any exit signal to ensure everything is cleaned up
+trap "shutdown; exit 1" ERR SIGINT SIGTERM
+
+# 1. Verify that the logs have the startup pattern (only if log file is provided)
+LOG_FILE=$1
+STARTUP_PATTERN="Block Node Server started at port"
+
+if [[ -n "$LOG_FILE" ]]; then
+    if grep -q "$STARTUP_PATTERN" "$LOG_FILE"; then
+        echo "Startup pattern found in logs."
+    else
+        echo "Startup pattern not found in logs."
+        exit 1
+    fi
+else
+    echo "No log file provided, skipping startup pattern check."
+fi
+
```

Comment:
Are you seeing output from the `consumer.sh` in this order? The consumers should only start seeing output at the beginning of the next full block (header blockitem). If we're only sending one block, then whether the consumer sees output would be a race condition. If we switch the order of #2 and #3 then the consumer *should* always see the whole block in their output.

Reply:
Thanks for the tip!

✅  Done!